File: solutions/lua/isbn-verifier/1/isbn-verifier.lua
return {
    valid = function(isbn)
        local count = 0
        local sum = 0
        for i = 1, #isbn do
            local c = isbn:sub(i, i)
            if (c >= '0' and c <= '9') then
                sum = sum + tonumber(c) * (10 - count)
                count = count + 1
            elseif (c == 'X' and count == 9) then
                sum = sum + 10
                count = count + 1
            elseif (c ~= '-') then
                return false
            end
        end
        return count == 10 and sum % 11 == 0
    end,
}

File: solutions/go/isbn-verifier/1/isbn_verifier.go
package isbn

func IsValidISBN(isbn string) bool {
	count := 0
	sum := 0
	for i := 0; i < len(isbn); i++ {
		if isbn[i] >= '0' && isbn[i] <= '9' {
			sum += int(isbn[i]-'0') * (10 - count)
			count++
			continue
		}
		if isbn[i] == 'X' && count == 9 {
			sum += 10
			count++
			continue
		}
		if isbn[i] != '-' {
			return false
		}
	}
	return count == 10 && sum%11 == 0
}
